### ISSUE TITLE
Changing hyper parameters during training based on scalars

### ISSUE THREAD

**Issue by vandenheuvel** [tensorflow/tensorflow#42370]
I'm training a model and logging values to tensorboard. Based on these values, I would like to tune hyperparameters (in code, not by hand). For example, one could think about switching to a different loss function, or changing a coefficient in the loss function, as a function of the loss.

It is not hard to find _some_ way to do it: one way would be to simply:

1. Work with eager execution
2. Save the scalars I'm logging to tensorboard also to a python list
3. Do computations on those lists in between batches
4. Execute different eager tensorflow logic, or assign different hyperparameters, based on those computation

Is there a way that is more idiomatic? For example, it would be nice to:

- Not have to save all these scalars to python lists myself. Can they be retrieved from tensorboard in a performant way?
- Use the tf.function api for better performance

This was originally a [Stack Overflow question](https://stackoverflow.com/questions/63403109/changing-hyper-parameters-during-training-based-on-scalars).

**Comment by gowthamkpr** [comment]
Are you using keras here? Can you please explain what api are you using here. Thanks!

**Comment by vandenheuvel** [comment]
I'm also using keras, but for this question, only tensorflow seems relevant. I'm just computing scalars all the time, such as losses, and logging those. I want to access all these historical loss values in a performant way, e.g. to determine whether the loss has stabilized (enough) or not.

**Comment by vandenheuvel** [comment]
Are these features (and is this documentation) new? https://www.tensorflow.org/tensorboard/dataframe_api

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
